### ISSUE TITLE
Validation Package - Email validation

### ISSUE THREAD

**Issue by ivanbacher** [aurelia/aurelia#1919]
https://github.com/aurelia/aurelia/blob/2fca6ea3703f51a2b7b87d719bfa45c9bb35d6e1/packages/validation/src/rule-provider.ts#L311

Not sure if this is the intended behavior.

At the moment the behaviour is like this

`test` -> not valid
`test@` -> not valid
`test@test`  -> valid
`test@test.` -> not valid
`test@test.c` -> valid

Should `test@test` also not be valid?

If yes, then changing the regex to something like this:
```
^[a-zA-Z0-9.!#$%&'*+\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*\.[a-zA-Z0-9]{2,}$
```

Should fix it.

**Comment by BBosman** [comment]
Technically `test@ai` is a valid email address. 

It's called a [dotless domain](https://en.wikipedia.org/wiki/Top-level_domain#Dotless_domains).

But they're rare and advised against by ICANN (among others), so you could indeed argue that calling them not valid makes more sense.

**Comment by Sayan751** [comment]
Note that this email pattern is similar to what is there for au1, and only provides a rudimentary email validation. An actual RFC 5322, and RFC 6532 compliant email address parser is a non-trivial job, and IMO is immensely difficult to handle only using regex (disclaimer: I wrote a [parser](https://github.com/Sayan751/email-address-parser)). I think that it is best to remove this OOTB rule from the validation package, so that devs can choose a fitting parser for email-address validation.

**Comment by aniketkumar7** [comment]
Yes, I think emailPattern does not enforce the presence of a top-level domain (TLD) at the end of the email address. This causes emails like test@test to be considered valid, even though they are missing a TLD (e.g., .com, .net, .org, etc.).

To fix this issue, the updated emailPattern adds \.[a-zA-Z0-9]{2,} at the end of the regular expression.
Is this enough? 
Can I contribute/change it?

**Comment by Sayan751** [comment]
> Can we use npm package validator.js to fix the issue?

@Swarga-codes The general idea is not to create dependency on third party library when it is avoidable. However, if there is a separate package that integrates the `@aurelia/validation` with a third party library like validator.js (something like `@aurelia/validation-validator-js`), I have nothing against that. @bigopon What do you think?

> Can I contribute/change it?

@aniketkumar7 Please feel free. Thank you for expressing your interest. However, as I have expressed before, my preference would be to remove the email rule, so that devs can choose a more competent email-address validation library as per their preference. An RFC compliant validation of email addresses is a very hard job for a regex.

**Comment by bigopon** [comment]
> integrates the @aurelia/validation with a third party library

We cannot cover all libraries, and providing an official one with a library will limit the potential of others, I think we should have a doc section for integration instead. It's harder to keep it up to date, but these libraries shouldn't change too fast, at least for now.

**Comment by Sayan751** [comment]
So we agree on removing the email rule 😛 

> I think we should have a doc section for integration instead.

We already have: https://docs.aurelia.io/aurelia-packages/validation/defining-rules#associating-validation-rules-with-property

![image](https://github.com/aurelia/aurelia/assets/3128881/785b666d-ff16-4345-9b67-426560ab4b9a)

**Comment by bigopon** [comment]
We had a precedence in v1 where someone argued hard enough to remove our default half-baked html sanitizer. For email, I don't think we need to be on the same level of strictness. I think we can have a good enough regex doing for the most common email patterns, and leave maliciously crafted email for the application servers to handle.
